test/unit/lodash.spec.js

```diff
@@ -0,0 +1,568 @@
+                expect(params[0][params[2]]).to.eql(params[1]);
+            }
+        );
+        it('should check if string starts with the given target string', function () {
+                expect(_.startsWith.apply(_, params)).to.eql(true);
+            }
+        );
+    });
+
+    describe('template', function () {
+        beforeEach(function () {
+            params = datasets.template();
+        });
+        it('should match types of passing elements', function () {
+            var param1 = params[0];
+            var param2 = params[1];
+            expect(param1).to.be.a('string');
+            expect(params).to.have.length.within(1, 2);
+            if (param2 !== undefined && params.length == 2) {
+                expect(param2).to.be.an('object');
+                expect(Object.keys(param2)).to.have.length.within(1, 6);
+            }
+        });
+        it('should contain specific option parameter', function () {
+            var param2 = params[1];
+            var options = ['escape', 'evaluate', 'imports', 'interpolate', 'sourceURL', 'variable'];
+
+            expect(param2).to.have.any.keys(options);
+            for (var key in param2) {
+                expect(key).to.be.oneOf(options);
+                if (key === options[0]) {
+                    expect(key).to.be.a('RegExp');
+                }
+                if (key === options[1]) {
+                    expect(key).to.be.a('RegExp');
+                }
+                if (key === options[2]) {
+                    expect(key).to.be.an('Object');
+                }
+                if (key === options[3]) {
+                    expect(key).to.be.a('RegExp');
+                }
+                if (key === options[4]) {
+                    expect(key).to.be.a('string');
+                }
+            }
+        });
+        it('should not contain empty option parameter', function () {
+                var param2 = params[1];
+                for (var key in param2) {
+                    expect(param2[key]).to.not.be.empty;
+                }
+            }
+        );
+    });
+
+    describe('toLower', function () {
+        before(function () {
+            params = datasets.toLower();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return converted string', function () {
+                expect(_.toLower.apply(_, params)).to.eql('some string to test function');
+            }
+        );
+    });
+
+    describe('toUpper', function () {
+        before(function () {
+            params = datasets.toUpper();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return converted string', function () {
+                expect(_.toUpper.apply(_, params)).to.eql('SOME STRING TO TEST FUNCTION');
```

Comment:
In my opinion, without looking to this file, that solution is impossible to reveal